pilot/pkg/serviceregistry/kube/controller/ambient/multicluster.go

```diff
@@ -107,12 +107,21 @@ func (a *index) buildGlobalCollections(
 				log.Warnf("Failed to sync gateways informer for cluster %s", c.ID)
 				return nil
 			}
+
+			// N.B we're not using the opts.WithXXX pattern here since we want to be very obvious about which
+			// stop is being used to shutdown the collection (it should always be the cluster stop, NEVER
+			// the top-level stop associated with the ambient controller)
+			opts := []krt.CollectionOption{
+				krt.WithName(fmt.Sprintf("ambient/GatewaysWithCluster[%s]", c.ID)),
+				krt.WithDebugging(opts.Debugger()),
```

Comment:
Is WithDebugging intended to be left in the code?

Reply:
Yes there's a krt Debugger instances that gets created in a new OptionsBuilder; it powers the debug/krtz endpoints for reading krt state. This code ensures that the debugger struct gets propagated even though we have a different (per-cluster) stop

Reply:
This explicit reference is only needed when the cluster stop is being overridden, otherwise the stop (and debugger) in opts is used

Reply:
Got it. Thank you